app/models/script.rb

```diff
@@ -0,0 +1,24 @@
+class Script < ApplicationRecord
+
+  has_many :events
+
+  before_validation :add_or_update_checksum
+
+  validates_presence_of :text
+  validate :unique_text
+
+  private
+
+  def add_or_update_checksum
+    return nil unless self.text # avoid triggering error on Script.new.valid?
+    checksum = Digest::SHA1.hexdigest(self.text)
+    if self.checksum.nil? || self.checksum != checksum
+      self.checksum = checksum
+    end
+  end
+
+  def unique_text
+    preexisting_text = self.class.where(checksum: self.checksum).first
+    errors.add(:script, "already exists.  Use script with uuid #{preexisting_text.id}") if preexisting_text
```

Comment:
Probably at some point we should consider moving all of these user facing strings to a `locale` file, in the off chance we want to serve non-english speaking API consumers (which may be specifically handy given the activist aims of ragtag).